### ISSUE TITLE
JS Module from unpkg is broken

### ISSUE THREAD

**Issue by danielweck** [sgtpep/csstag#2]
`import css from 'https://unpkg.com/csstag';` should be `import css from 'https://unpkg.com/csstag?module';`, and there should be a `module` field in `package.json`.

Example to follow:
https://unpkg.com/browse/htm@3.0.3/package.json

Cheers, and thanks for this project :)
(just toying with it at the moment)

**Comment by danielweck** [comment]
My current workaround: `import css, { appendStyles } from 'https://unpkg.com/csstag/index.mjs';`

**Comment by sgtpep** [comment]
Hey, Daniel! Thanks for sharing this tip with me! I've just added "module" key to package.json for that (haven't published the new version, though, yet). But I also found out that the `import css from 'https://unpkg.com/csstag'` seems to work just fine in the demo https://sgtpep.github.io/csstag/demo/.

**Comment by danielweck** [comment]
Yes, `import css from 'https://unpkg.com/csstag'` is a HTTP 302 redirect to `https://unpkg.com/csstag@0.3.0/index.min.js` (UMD, I am guessing?).

`'https://unpkg.com/csstag?module';` should be a HTTP 302 redirect to the ESM file `https://unpkg.com/csstag@0.3.0/index.mjs` (adding the `module` field in `package.json` should fix that).

The `?module` query param is automatically "forwarded" by `unpkg` in the 302 redirect URL and into subsequent HTTP requests (i.e. cascaded module dependencies). For example, if the `csstag` NPM package code contained Javascript ESM imports such as `import bar from 'foo';`, the `foo` "bare" module specifier would automatically be resolved/expanded to a `unpkg` URL with the `?module` query param. Note that this CDN feature is currently experimental: https://unpkg.com/#query-params

Here is a concrete example with the Preact Hooks module:

https://unpkg.com/preact@10.3.4/hooks/dist/hooks.module.js
(note the line of code at the top: `import{options as n}from"preact";`)
versus:
https://unpkg.com/preact@10.3.4/hooks/dist/hooks.module.js?module
(note the line of code at the top: `import { options as n } from "https://unpkg.com/preact@latest?module";`)

In some hypothetical future, `csstag` could leverage this feature in its `index.mjs` Javascript ESM module by importing the `postcss` dependencies via ESM resolved/served directly by `unpkg` ... but at this point in time these use Node/CommonJS `require()` imports, so the module dependency tree/graph does not support ESM out-of-box:
https://unpkg.com/browse/postcss-modules-local-by-default@3.0.2/index.js
https://unpkg.com/browse/postcss-modules-parser@1.1.1/lib/index.js
https://unpkg.com/browse/postcss-modules-scope@2.2.0/src/index.js
https://unpkg.com/browse/postcss-modules-scope@2.2.0/src/index.js
https://unpkg.com/browse/postcss-modules-values@3.0.0/src/index.js
=> PostCSS:
https://unpkg.com/browse/postcss@7.0.27/lib/postcss.js
etc.

I see that `csstag`'s Rollup config is already using https://github.com/rollup/plugins/tree/master/packages/commonjs :
https://github.com/sgtpep/csstag/blob/master/rollup.config.js#L1
...so in principle, it should be possible to externalize the PostCSS dependencies out of `csstag`'s`index.mjs`, which would optimize delivery/caching to web browser module consumers.
By the way, here's some useful information regarding the CJS -> ESM transform (this is a third-party alternative to the Rollup plugin):
https://github.com/eight04/rollup-plugin-cjs-es#comparing-to-rollup-plugin-commonjs

Just thoughts :)

PS: at some point I'll be trying `csstag`'s Babel compiler in order to remove the runtime overhead, but at the moment I am just experimenting with a few Preact + Htm + CssTag SPA techniques for a pet project (in-browser pure-client-side no-transpilation-step live-reload quick-prototyping). `htm` also has a Babel compiler https://github.com/developit/htm , which would leave only Preact as a runtime dependency.

**Comment by danielweck** [comment]
Ah, I've just realized the inconsistency in `csstag`'s Rollup build output:
https://github.com/sgtpep/csstag/blob/7d8d785bcdbfeb8463c3878cadcc6985998c593f/rollup.config.js#L20-L37
`index.min.js` (ESM) is *not* a minified version of `index.js` (CJS), it is in fact the minified version of `index.mjs` (ESM).

...thus my confusion as to why `import css from 'https://unpkg.com/csstag'` worked out of the box :)

**Comment by danielweck** [comment]
Note that the Pika CDN cannot find the `module` entry in `package.json` (this should be fixed once an update to the `csstag` NPM package is published):
https://www.pika.dev/search?q=csstag

**Comment by danielweck** [comment]
Another benefit of adding the `module` property in `package.json` is that tooling like `snowpack` will support `csstag`:
https://www.snowpack.dev/#zero-config-installs-(default)

**Comment by sgtpep** [comment]
Thanks for the elaborated explanation! It definitely worth adding `"module"`. I published the new version with it, so that https://unpkg.com/csstag?module now redirects properly.

**Comment by sgtpep** [comment]
> PS: at some point I'll be trying csstag's Babel compiler in order to remove the runtime overhead, but at the moment I am just experimenting with a few Preact + Htm + CssTag SPA techniques for a pet project (in-browser pure-client-side no-transpilation-step live-reload quick-prototyping). htm also has a Babel compiler https://github.com/developit/htm , which would leave only Preact as a runtime dependency.

That was exactly the use-case and 'stack' `csstag` was designed for. I didn't have time to really promote this approach or `csstag` itself, so it might look a bit abandoned. But I'm unaware of alternatives among css-in-js in this 'niche', so maybe it worth giving this project another boost.

**Comment by danielweck** [comment]
Thanks a bunch for your prompt responses! :)
I will let you know if I encounter other developer experience niggles, or runtime bugs.
Right now I envision using `csstag` for quick and friction-free prototyping, but will use the Babel transform to eliminate the costly PostCSS runtime in production builds (same design approach with `htm`).

**Comment by danielweck** [comment]
(ah, sorry for repeating myself in the above comment, I posted just after you did :)

**Comment by danielweck** [comment]
> Thanks for the elaborated explanation! It definitely worth adding `"module"`. I published the new version with it, so that https://unpkg.com/csstag?module now redirects properly.

I think this should serve the minified JS module (i.e. huge bundle with PostCSS).

For example:

MODULE:
https://unpkg.com/htm@latest?module
REDIRECTS TO:
https://unpkg.com/htm@3.0.3/dist/htm.module.js?module

UMD:
https://unpkg.com/htm@latest
REDIRECTS TO:
https://unpkg.com/htm@3.0.3/dist/htm.js

This is why I mentioned the confusion (in a previous comment) related to `csstag`'s current naming conventions (i.e. `index.min.js` (ESM) is not a minified version of `index.js` (CJS), it is in fact the minified version of `index.mjs` (ESM)), and I also suggested providing a UMD dist.

For the "ECMAScript module" build, I would suggest using `*.module.js` or `*.esm.js` instead of `*.mjs` as the `*.js` file extension is supported everywhere.
For the CommonJS build, I would suggest using `*.cjs.js`. 
For the UMD build, I would suggest using `*.js`.
For each of the 3 dist targets, I would use `*.min.*` for the minified bundles, and I would reference them for the UMD and ESM fields in `package.json` (I think it is okay to point to the non-minified JS for CommonJS / Node, as minification would typically be handled by some tooling further downstream)

**Comment by sgtpep** [comment]
Hey, Daniel! Sorry for the late reply, had a rough week. Thank you for such an elaborated recommendation.

I've updated the package adding index.cjs.min.js, index.esm.min.js and extra package.json fields to it. I had issues building UMD target. Is this target still relevant with ESM becoming mainstream?

Also I've noticed an interesting thing: if package.json has "browser" field, then unpkg prefers it over "main" on https://unpkg.com/csstag.

**Comment by danielweck** [comment]
Thank you for the updates! Sorry for the late reply, I have been pretty busy :)

I have made a "fork" of CssTag so that it integrates more seamlessly in my project. I am keeping a close eye on your "upstream", thank you so much for creating CssTag!

I will contribute back any useful addition/modification. At the moment my project is closed-source, but I have only modified the build aspects of CssTag: notably I tweaked the Babel pipeline to enable a downgrade path for IE11 (via a global Rollup config and PostCss auto-prefixer), and I also have Stylelint integration (which forced me to implement a different syntax for `@value`, and I inserted a Babel `visitor.Program.exit` stage to handle the conversions).

Fun stuff! :)
